pkg/services/ngalert/state/historian/loki.go

```diff
@@ -248,7 +248,12 @@ func statesToStreams(rule history_model.RuleMeta, states []state.StateTransition
 		labels[RuleUIDLabel] = fmt.Sprint(rule.UID)
 		labels[GroupLabel] = fmt.Sprint(rule.Group)
 		labels[FolderUIDLabel] = fmt.Sprint(rule.NamespaceUID)
-		repr := labels.String()
+		lblJsn, err := json.Marshal(labels)
```

Comment:
Choice of JSON as a stable representation was somewhat arbitrary.
- Hash of labels also works and is more compact. But, we're also relying on the labels being parseable later on, so we'd need to store the labels again separately, which undoes any memory savings. 
- `data.LabelsFromString(...)` already [understands JSON](https://github.com/grafana/grafana-plugin-sdk-go/blob/cdc5808e01ce7d8c5db8665207b0035bbb1fd82c/data/labels.go#L91-L96), so no code changes needed on parse side.